modules/library/main/src/main/java/org/geotools/filter/function/FilterFunction_mapGet.java

```diff
@@ -0,0 +1,75 @@
+/*
+ *    GeoTools - The Open Source Java GIS Toolkit
+ *    http://geotools.org
+ *
+ *    (C) 2020, Open Source Geospatial Foundation (OSGeo)
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License as published by the Free Software Foundation;
+ *    version 2.1 of the License.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ */
+package org.geotools.filter.function;
+
+import static org.geotools.filter.capability.FunctionNameImpl.parameter;
+
+import java.util.Map;
+import org.geotools.factory.CommonFactoryFinder;
+import org.geotools.filter.FunctionExpressionImpl;
+import org.geotools.filter.capability.FunctionNameImpl;
+import org.geotools.util.Converters;
+import org.opengis.filter.FilterFactory2;
+import org.opengis.filter.capability.FunctionName;
+import org.opengis.filter.expression.PropertyName;
+import org.opengis.filter.expression.VolatileFunction;
+
+/** Extracts a value from a map given the mapname.key identifier as a parameter */
+public class FilterFunction_mapGet extends FunctionExpressionImpl implements VolatileFunction {
+
+    FilterFactory2 FF = CommonFactoryFinder.getFilterFactory2();
+
+    public static FunctionName NAME =
+            new FunctionNameImpl(
+                    "mapGet",
+                    parameter("value", Object.class),
+                    parameter("mapWithKey", String.class));
+
+    public FilterFunction_mapGet() {
+        super(NAME);
+    }
+
+    @Override
+    public Object evaluate(Object object, Class context) {
+        Object result = evaluate(object);
+        if (result == null) {
+            return null;
+        } else {
+            return Converters.convert(result, context);
+        }
+    }
+
+    public Object evaluate(Object feature) {
+        String name = getExpression(0).evaluate(feature, String.class);
+
+        if (name == null) {
+            return null;
+        }
+
+        String[] split = name.split("\\.");
```

Comment:
Normally access to property values is done via dedicated property support, e.g., it could be:

``mapGet("myProperty", 'mapKey')``

Where "myProperty" becomes a PropertyName... if it's dynamic, it could also be:

``mapGet(property('myProperty'), mapKey);

At the very least, I would check if the key has a dot in the middle, and if not, evaluate directly assuming the "feature" is a map.

Reply:
fixed. thanks